### ISSUE TITLE
Trim Audio can erroneously delete other unselected clips to the left (or right)

### ISSUE THREAD

**Issue by petersampsonaudacity** [audacity/audacity#4165]
### Bug description

If a selection is made in a clip that touches (or overlaps) the left (or right) boundaries of the clip the when using **Edit > Remove Special > Trim Audio** or its shortcut or button then erroneously all clips to the left (or right) in the track.

The STR use case is a simple one with just 3 clips but in a multi-clip project **All** clips to the left (or right) of the selected clip are also deleted.

If an entire clip is selected and Trim is used the all clips to the left and right of the selected clip are deleted.

This also happens if you make a selection with multiple clips where one of the selection boundaries is on the boundary edge of one of the clips.

Note that it states (and has always stated) quite clearly in the Manual:
>Removes all audio from the current [clip](https://alphamanual.audacityteam.org/man/Audacity_Tracks_and_Clips) except the selected part, creating its own clip from the remaining audio. 
>If there are other separate clips in the same track these are not removed or shifted.

This does _**not**_ appear to be a regression as this same behavior can be observed in all versions back to and including 2.0.0 

### Steps to reproduce

1. get some audio
2. select a section in the middle
3. **Ctrl + I**  to split to produce 3 clips
4. move the clips so that there is empty space between them
5. Make a selection in the middle clip such the the selection touches or overhangs the left (or right) clip boundary
6. **Ctrl + T** to trim the middle clip  (or command or Edit Toolbar button)
7. Observe: middle clip is trimmed _**BUT**_ clip to the left (or right) is also erroneously removed

### Expected behavior

Clips to the left (or right) of a clip should  _**not**_ be removed when Trim is used on a clip or clips selection that touches the boundary of one of the clips

### Actual behavior

See above.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

Originally commented on in the Manual by @billwharrie when he was updating the Reference Manual for "smart clips".  
https://alphamanual.audacityteam.org/man/Edit_Menu:_Remove_Special#trim_audio

He added it as an alert div - I believe it to be a long-standing bug:
![image](https://user-images.githubusercontent.com/50205984/211532754-d0765c75-ffe5-471e-8114-c9574d6f9e29.png)

**Comment by petersampsonaudacity** [comment]
I have updated the Manual's alert div:
![image](https://user-images.githubusercontent.com/50205984/211535722-caaa3659-5b83-4158-8718-1bf1f7ed2b24.png)

**Comment by DavidBailes** [comment]
> Note that it states (and has always stated) quite clearly in the Manual: https://alphamanual.audacityteam.org/man/Edit_Menu:_Remove_Special#trim_audio
> 
> > Removes all audio from the current [clip](https://alphamanual.audacityteam.org/man/Audacity_Tracks_and_Clips) except the selected part, creating its own clip from the remaining audio.
> > If there are other separate clips in the same track these are **not removed or shifted**.

I think that the manual is incorrect. It appears that the behaviour of trim was changed in 2005 by this commit:

Author:			nestify <nestify@47890b92-0858-11df-a26f-8b716316a5bc>
Date:			17 years ago (16/10/2005 08:29:21)
Commit hash:	466ddd5fe51b14cb4abef281e73aa81b39f114a1
Child:			66252862
Parent:			56147897

Changing 'trim' function so that it trims the current clip, or everything outside a selection bound that isn't in a clip.
Contained in branches:
master

**Comment by petersampsonaudacity** [comment]
> Changing 'trim' function so that it trims the current clip, or everything outside a selection bound that isn't in a clip.

If that were fully the case then I would expect Trim Audio to remove everything in a track outside the current selection.

But my steps to reproduce show that this is _**not**_ the case - the STR use cases above depend entirely on a selection abutting at least one of the boundary edges of a clip (or overlapping the edge into empty space in the track)

Applying Trim Audio to this selection:
![image](https://user-images.githubusercontent.com/50205984/211563976-795a4f60-e1e8-426c-afb6-371c92d09d1e.png)

Produces this:
![image](https://user-images.githubusercontent.com/50205984/211564190-4d46f20d-25bb-4a0c-b29f-42e4e4b2a5bd.png)
Note that _**only**_ the Clip is trimmed to the selection and _**not**_ the whole track.

And this is _**precisely**_ what it says (and has long said in the Manual, wher it says it operates on the "clip" it does ### _not_ say it operates on the "track" - and I believe this to be desirable/preferred bhavior

**Comment by petersampsonaudacity** [comment]
In contrast

Selecting this
![image](https://user-images.githubusercontent.com/50205984/211565083-b6e69df7-6567-48d5-9064-f85267b291b8.png)

And then applying Trim Audio produces this:
![image](https://user-images.githubusercontent.com/50205984/211565597-313da635-178a-42bc-89b3-c3570cde12ac.png)

Which is the behavior I reported in this issue thread

**Comment by SteveDaulton** [comment]
In the light of David's comment above, it seems clear that this is an intended behaviour (a "feature" rather than a "bug"), though it does seem to me to be weirdly and confusingly inconsistent to have one (and only one) command that treats _selections ending on a clip boundary_ differently from any other  selection.

If it were down to me, and in the absence of any obvious use case for the weirdness, I'd make "Trim" apply consistently to "clips".

**Comment by petersampsonaudacity** [comment]
> Before there were clips,

But we've had clips ever since the early days of 1.3.0 ...

And the initial page in the manual on Trim had this to say (on 25Jan2013):
https://alphamanual.audacityteam.org/m/index.php?title=Edit_Menu:_Remove_Special&oldid=22842

>Removes all audio from the current [clip](https://alphamanual.audacityteam.org/man/Audacity_Tracks_and_Clips) 
>except the selected part, creating its own clip from the remaining audio. 
>If there are other separate clips in the same track these are not removed or shifted 
>unless trimming the entire length of a clip or clips (note in the "After" image below that the audio in the clip to the right is unaffected).

The part that says "unless trimming the entire length of a clip or clips ..." was removed on 30Jul17 as part of a "Tweak descriptions for clarity" change.

-----------------------------------------------------------------


But I **_really_** think that Trim should work on **Clips** and not on tracks - especially since clips are now a pre-eminent part of Audacity.

And thus I believe that the behavior outlined in the STR should be fixed - if not we have behaviot that is oddly inconsistent and is not easily "discoverable".

**Comment by petersampsonaudacity** [comment]
> After all this, I agree with Steve. My design choice would be: If an entire clip is selected nothing happens (which is consistent with the idea that Trim acts on clips, not tracks), otherwise the un-selected audio in the clip is deleted.

Big Pus 1 to that

**Comment by SteveDaulton** [comment]
> But we've had clips ever since the early days of 1.3.0 ...

But this "feature" was added before 1.3.0, so either it was added in the "stable" branch at around 1.2.3, or in the "beta" branch before the first release of the 1.3.x branch. I don't recall exactly when "clips" were first introduced, but there were no "clips" in Audacity 1.0.0, just "tracks".

**Comment by billw58** [comment]
The Audacity 1.2 manual is available on the web, and it shows the Trim command under the Edit menu. I distinctly remember switching from 1.2.8 (?) (the last stable 1.2.x for Mac) to the 1.3.x "beta" when clips became available.

Anyway, it will be up to Muse to decide what the UI should do.

**Comment by petersampsonaudacity** [comment]
Let me give you a thought-experiment use-case that shows why that fixing this as outlined in the bug report is **_important_** (rather than relying on an odd "design decision" committed by an external unknown developer "**nestify**" 17 years ago.

1. User has a track that has say 25 clips in it _(not unusual from some user projects I have seen)_
2. User wants to edit the first clip, say to remove and fade-in overlong applause 
3. For greater accuracy user zooms in to the first clip so that it (and it alone) fits on the screen)
4. user identifies the entry point and selects from there to the end of the first clip
5. User then uses the Trim Audio command
6. User then fades in and maybe makes some other edits to the first clip
7. User the zooms to fit and is surprised to see that they now have hsut the trimmed/faded/edited first clip.

How on earth is the average user supposed to intuit or know that at step 5 they are also removing/deleting the other 24 clips in the track?

Sure they can use Undo to restore these = but to get to that particular undo they also have to undo the edits at step 6, of which there may be several.

Tagging @Tantacrul @DilsonsPickles and @LWinterberg   -  as this is a design/UX issue

**Comment by DavidBailes** [comment]
> an external unknown developer "**nestify**"

just for the record, nestify was Shane T Mueller's username.

**Comment by Tantacrul** [comment]
Agreed. It should only apply to the selected clip (or clips) and no others. Can't see an advantage to the existing behaviour.

I'd also add that the trim option is ridiculously buried - to the point where it's tough to believe anyone would ever find it.

**Comment by LWinterberg** [comment]
fwiw, the trim option has a dedicated button in addition to its menu entry.

**Comment by SteveDaulton** [comment]
> I'd also add that the trim option is ridiculously buried - to the point where it's tough to believe anyone would ever find it.


![edit_toolbar_3_2_3](https://user-images.githubusercontent.com/11681097/211809456-4eec4529-0ea0-43e5-a77b-1661446d543c.png)

**Comment by Tantacrul** [comment]
Ah, my mistake. I thought it was a special, niche function, since it was in a menu called 'Remove special'. Didn't realise it was this!

**Comment by Ryan-Miller-Audacity** [comment]
Testing against the latest public release (Audacity 3.3.2) this is still reproducible there and can be discussed in our next triaging session.
